src/main/java/com/activestate/cloudfoundryjenkins/CloudFoundryPushPublisher.java

```diff
@@ -726,4 +763,20 @@ public FormValidation doCheckHostname(@QueryParameter String value) {
             return FormValidation.validateRequired(value);
         }
     }
+    /**
+     * Adds a route to an existing application.
+     * @param client
+     * @param appName
+     * @param newRoute
+     */
+    public static void addRouteToApplication(CloudFoundryClient client, String appName, String newRoute) {
+        try {
+            System.out.println("Adding route " + newRoute + " to " + appName);
+            List<String> routes = client.getApplication(appName).getUris();
+            routes.add(newRoute);
+            client.updateApplicationUris(appName, routes);
+        } catch (CloudFoundryException e) {
+            System.out.println("Application " + appName + " was not found while trying to add route.");
```

Comment:
remove sops.
Can't there be any other error if CloudFoundryException is thrown ?